### PR TITLE
Implement unarchive feature

### LINKED ISSUE #75: As a store manager, I can unarchive/restore back an archived empolyees' contact details
... so that I can add the employee contact information back to my active contact list when I rehire them.

### PR DESCRIPTION
**Pull request by yanbingtao**
Allows users to unarchive a contact detail from the archived list back to active list.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/UnarchiveCommand.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.ArchiveStatus;
+import seedu.address.model.person.Person;
+
+
+
+
+public class UnarchiveCommand extends Command {
```
Comment (WM71811) [inline src/main/java/seedu/address/logic/commands/UnarchiveCommand.java]: can add a javadoc here?

File: src/main/java/seedu/address/logic/parser/UnarchiveCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.UnarchiveCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+
+public class UnarchiveCommandParser {
```
Comment (WM71811) [inline src/main/java/seedu/address/logic/parser/UnarchiveCommandParser.java]: similarly javadoc can be added..?

File: src/main/java/seedu/address/model/person/ArchiveStatus.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+
+public class ArchiveStatus {
```
Comment (WM71811) [inline src/main/java/seedu/address/model/person/ArchiveStatus.java]: similar as above

File: src/main/java/seedu/address/logic/commands/UnarchiveCommand.java (hunk 1 above)
Comment (yanbingtao) [reply src/main/java/seedu/address/logic/commands/UnarchiveCommand.java]: Sure, I've added that. Thanks!

File: src/main/java/seedu/address/logic/parser/UnarchiveCommandParser.java (hunk 2 above)
Comment (yanbingtao) [reply src/main/java/seedu/address/logic/parser/UnarchiveCommandParser.java]: Sure, I've added that. Thanks!

File: src/main/java/seedu/address/model/person/ArchiveStatus.java (hunk 3 above)
Comment (yanbingtao) [reply src/main/java/seedu/address/model/person/ArchiveStatus.java]: Sure, I've added that. Thanks!

**Review by WM71811 — DISMISSED** [review]
LGTM

File: src/test/java/seedu/address/testutil/PersonBuilder.java [hunk 4]
```diff
@@ -18,10 +19,12 @@
     public static final String DEFAULT_NAME = "Alice Pauline";
     public static final String DEFAULT_PHONE = "85355255";
     public static final String DEFAULT_ADDRESS = "123, Jurong West Ave 6, #08-111";
+    public static final boolean DEFAULT_ARCHIVESTATUS = false;
 
     private Name name;
     private Phone phone;
     private Address address;
+    private ArchiveStatus archiveStatus;
```
Comment (Diwu-Yi) [inline src/test/java/seedu/address/testutil/PersonBuilder.java]: Perhaps the naming of this field could be changed to isArchived to show that ArchiveStatus can only be true or false ?

Comment (yanbingtao) [reply src/test/java/seedu/address/testutil/PersonBuilder.java]: But I think normally for variable names, we should use a noun. And only for method, we use something starting with a verb, like isArchived. So I think here, archiveStatus should be used (: Thanks!

**Review by WM71811 — DISMISSED** [review]
LGTM

File: src/main/java/seedu/address/logic/commands/UnarchiveCommand.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+/**
+ * Unarchives a person identified using it's displayed index from the address book.
+ */
+public class UnarchiveCommand extends Command {
+
+    public static final String COMMAND_WORD = "c-unarchive";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Unarchive the person identified by the index number used in the displayed person list.\n"
```
Comment (ureshiiYing) [inline src/main/java/seedu/address/logic/commands/UnarchiveCommand.java]: Should be `Unarchives` as the command is singular

File: src/main/java/seedu/address/logic/commands/UnarchiveCommand.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+     */
+    public UnarchiveCommand(Index targetIndex) {
+        this.targetIndex = targetIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
```
Comment (ureshiiYing) [inline src/main/java/seedu/address/logic/commands/UnarchiveCommand.java:45]: Is this an extra line?

File: src/main/java/seedu/address/model/person/Person.java [hunk 7]
```diff
@@ -77,12 +77,23 @@ public boolean isSamePerson(Person otherPerson) {
     /**
      * Sets the person's archive status to true. It's equivalent to archive the person.
```
Comment (ureshiiYing) [inline src/main/java/seedu/address/model/person/Person.java]: Grammar: It is equivalent to *the person being archived) OR (having archived the person).

File: src/main/java/seedu/address/model/person/Person.java [hunk 8]
```diff
@@ -77,12 +77,23 @@ public boolean isSamePerson(Person otherPerson) {
...
      *
-     * @return A Person type with "true" archive status.
+     * @return A Person type with "true" in archive status.
```
Comment (ureshiiYing) [inline src/main/java/seedu/address/model/person/Person.java]: Could be better phrased as: "A Person whose archive status is true"

File: src/main/java/seedu/address/model/person/Person.java [hunk 9]
```diff
@@ -77,12 +77,23 @@ public boolean isSamePerson(Person otherPerson) {
...
      */
     public Person archive() {
         return new Person(this.name, this.phone, this.address, new ArchiveStatus(true),
                 this.tags);
     }
+
+    /**
+     * Sets the person's archive status to false. It's equivalent to unarchive the person.
```
Comment (ureshiiYing) [inline src/main/java/seedu/address/model/person/Person.java:88]: Similar to the above comment

**Review by ureshiiYing — CHANGES_REQUESTED** [review]
Overall, code logic seems good. There are just some grammatical errors to fix.

File: src/test/java/seedu/address/testutil/PersonBuilder.java (hunk 4 above)
Comment (Diwu-Yi) [reply src/test/java/seedu/address/testutil/PersonBuilder.java]: okay, sure

File: src/main/java/seedu/address/logic/commands/UnarchiveCommand.java (hunk 5 above)
Comment (yanbingtao) [reply src/main/java/seedu/address/logic/commands/UnarchiveCommand.java]: Thanks for pointing out, will change it.

File: src/main/java/seedu/address/logic/commands/UnarchiveCommand.java (hunk 6 above)
Comment (yanbingtao) [reply src/main/java/seedu/address/logic/commands/UnarchiveCommand.java:45]: yup, will delete this line away, thanks!

File: src/main/java/seedu/address/model/person/Person.java (hunk 7 above)
Comment (yanbingtao) [reply src/main/java/seedu/address/model/person/Person.java]: Will change this part, thank you!

File: src/main/java/seedu/address/model/person/Person.java (hunk 8 above)
Comment (yanbingtao) [reply src/main/java/seedu/address/model/person/Person.java]: Yes! that will be better, thanks!

File: src/main/java/seedu/address/model/person/Person.java (hunk 9 above)
Comment (yanbingtao) [reply src/main/java/seedu/address/model/person/Person.java:88]: Thank you!

**Review by WM71811 — APPROVED** [review]
LGTM

**Review by ureshiiYing — APPROVED** [review]
LGTM

**Pull request merged**